harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -338,17 +328,22 @@ def _aggregate_training_metrics(self, training_metrics: List[Dict]) -> Dict:
             )
 
         if not self.is_chief:
+            self.context._maybe_reset_tbd_writer()
             return {}
 
         # Only report on the chief worker
         avg_metrics = metrics.get("avg_metrics", {})
         batch_metrics = metrics.get("batch_metrics", [])
 
-        self.metric_writer.on_train_step_end(
+        self._log_tb_metrics(
+            "train",
             self.state.batches_trained,
             avg_metrics,
             batch_metrics,
         )
+
+        self.context._maybe_reset_tbd_writer()
```

Comment:
it makes sense to me that these should live in the `_upload_tb_files()` function.  There's no reason to reset the writer at any other time; neither does it make sense to _not_ reset right before any given upload.